sources/osgShadow/ShadowTextureAtlas.js

```diff
@@ -20,10 +20,10 @@ var ShadowTextureAtlas = function() {
     this._uniforms = {};
     this._lightNumberArray = []; // default for a valid cloneType
 
-    this._viewMatrices = [];
-    this._projectionMatrices = [];
-    this._depthRanges = [];
-    this._mapSizes = [];
+    this._viewMatrices = {};
```

Comment:
why switching [] to {}

Reply:
As we're now indexing by lightnumbers now, not using lightArrayIndex, so it's a sparse array
if user put lights 8,9,10, better use an associative array ?

Reply:
no idea what is better in this case, but simpler is probably better

Reply:
I would probably keep array because size will not be more up to 8 I guess